### PR TITLE
feat: support for OIDC providers in identitySource

### PR DESCRIPTION
**Pull request by reste85**
Adding support for OIDC providers in IdentitySource Construct

### CODE REVIEW

File: test/policy-store.test.ts [hunk 1]
```diff
@@ -661,4 +661,31 @@ describe('generating schemas from OpenApi specs', () => {
...
+      },
+    });
+
+    // THEN
+    expect(pStore.schema?.cedarJson).toBeDefined();
+    expect(Object.keys(schema.PodcastApp.entityTypes)).toStrictEqual([
+      'User',
+      'Application',
+    ]);
+    // it should have the eight explicitly defined actions plus the 6 derived from the 'any' definition
```
Comment (svenjaraether) [inline test/policy-store.test.ts:688]:
```suggestion
    // it should have the eight explicitly defined actions plus the six derived from the 'any' definition
```

File: test/policy-store.test.ts [hunk 2]
```diff
@@ -661,4 +661,31 @@ describe('generating schemas from OpenApi specs', () => {
     // it should have the eight explicitly defined actions plus the 6 derived from the 'any' definition
```
Comment (svenjaraether) [inline test/policy-store.test.ts:661]:
```suggestion
    // it should have the 8 explicitly defined actions plus the 6 derived from the 'any' definition
```

Comment (reste85) [reply test/policy-store.test.ts:661]: i would keep "eight" and "six" IMHO

File: test/policy-store.test.ts (hunk 1 above)
Comment (reste85) [reply test/policy-store.test.ts:688]: i would keep "eight" and "six" IMHO

**Review by svenjaraether — APPROVED** [review]
approved, as discussed

**Pull request merged**